cmake/modules/RootNewMacros.cmake

```diff
@@ -963,10 +958,18 @@ function(ROOT_STANDARD_LIBRARY_PACKAGE libname)
     set(ROOT_LIBRARY_TARGETS "${ROOT_LIBRARY_TARGETS};${libname}" CACHE STRING "List of ROOT targets generated from ROOT_STANDARD_LIBRARY_PACKAGE()" FORCE)
   endif()
 
+  if (PROJECT_NAME STREQUAL ROOT)
+    include_directories(BEFORE "inc")
+    if(IS_DIRECTORY "v7/inc")
+      include_directories(BEFORE "v7/inc")
```

Comment:
See my comment below on why we need to improve `ROOT_GENERATE_DICTIONARY` first to know which library target to pick up the include dirs from. I.e. this won't happen as part of this PR.

Reply:
It still makes sense to use `target_include_directories()` here on the library, plus add the include with `DICTIONARY_OPTIONS -I${incdir}` so that `rootcling` sees it. That's what I've been doing. See [`sql/mysql/CMakeLists.txt`](https://github.com/root-project/root/blob/master/sql/mysql/CMakeLists.txt) for an example.